client/src/components/FilterList.js

```diff
@@ -0,0 +1,239 @@
+import React, { useEffect, useState } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import Collapse from "@material-ui/core/Collapse";
+import ExpandLess from "@material-ui/icons/ExpandLess";
+import ExpandMore from "@material-ui/icons/ExpandMore";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Checkbox from "@material-ui/core/Checkbox";
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    width: "100%",
+    maxWidth: 360,
+    backgroundColor: theme.palette.background.paper
+  },
+  nested: {
+    paddingLeft: theme.spacing(4)
+  }
+}));
+
+export default function FilteredList(props) {
+  const classes = useStyles();
+  const [open, setOpen] = useState({
+    product: true,
+    price: false
+  });
+
+  const [products, setProducts] = useState({
+    all: true,
+    cakes: false,
+    cookies: false,
+    cupcakes: false,
+    macarons: false
+  });
+
+  const [pricing, setPricing] = useState({
+    all: true,
+    lessTen: false,
+    tenTwenty: false,
+    fourty: false,
+    fourtyMore: false
+  });
```

Comment:
This looks a lot better, however what if you structured it like this:
pricing was an array of ids that were checked, ids could be like "all", "lessTen", .. etc.

Then you could have all the product types for example in an object like this:
PRODUCT_TYPES = [{id: "all", label: "All", ... }]
But don't have the checked information in PRODUCT_TYPES, instead, its just a constant outside of the component

And then when you generate the components, you can just map PRODUCT_TYPES and check if the current id is in the pricing array to see if it's checked

To generate the API call then is really easy as well, since the array is all that you would really be passing to the API, etc.

Reply:
I tried this method. I am getting an error that checkbox checked expects a boolean but string was returned. I will commit this one and put this as a backlog for the next ticket.